### PR TITLE
applications: nrf_desktop: Improve simulated motion documentation

### PR DESCRIPTION
**Pull request by MarekPieta**
Change introduces additional information related to the implementation details of the module.

Jira: NCSDK-22790

### CODE REVIEW

File: applications/nrf_desktop/doc/motion.rst [hunk 1]
```diff
@@ -106,10 +116,15 @@ In these configurations, the module is a configuration channel listener and it p
 Implementation details
 **********************
 
+Implementation details depend on the selected implementation.
+
+Movement data from motion sensors
+=================================
+
```
Comment (mkapala-nordic) [inline applications/nrf_desktop/doc/motion.rst:123]: nit, can be ignored: I think that this selected and changing the headers related to motion sensor and heading style changes from line 127;136;147 could have been done in separate commit. Those changes relates to Motion Sensor module, not Motion Simulated one

**Review by mkapala-nordic — APPROVED** [review]

File: applications/nrf_desktop/doc/motion.rst (hunk 1 above)
Comment (MarekPieta) [reply applications/nrf_desktop/doc/motion.rst:123]: I modified the headers' ranks, because previously the section focused only on the motion sensor implementation and currently we want to have descriptions for other implementations next to it (introducing this change separately may not be best approach, because it's not needed unless description for the second implementation is introduced here).

**Comment by MarekPieta** [comment]
Please do not review yet. I am modifying the PR to introduce information related to motion buttons too.
We can proceed with the review as soon as I finish this one.

File: applications/nrf_desktop/doc/motion.rst [hunk 2]
```diff
@@ -141,12 +156,34 @@ Upon connection, the following happens:
...
     #. Samples the motion data.
-    #. Submits the ``motion_event``.
-    #. Waits for the indication that the ``motion_event`` data was transmitted to the host.
-       This is done when the module receives the ``hid_report_sent_event`` event.
+    #. Submits the :c:struct:`motion_event`.
+    #. Waits for the indication that the :c:struct:`motion_event` data was transmitted to the host.
+       This is done when the module receives the :c:struct:`hid_report_sent_event` event.
 
-#. At that point, a next motion sampling is performed and the next ``motion_event`` sent.
+#. At that point, a next motion sampling is performed and the next :c:struct:`motion_event` sent.
```
Comment (annwoj) [inline applications/nrf_desktop/doc/motion.rst:163]:
```suggestion
#. At this point, a next motion sampling is performed and :c:struct:`motion_event` is sent.
```

File: applications/nrf_desktop/doc/motion.rst [hunk 3]
```diff
@@ -141,12 +156,34 @@ Upon connection, the following happens:
...
+Accuracy of the generated movement data
+---------------------------------------
+
+The accuracy of generated simulated movement data depends on accuracy of the available hardware cycle counter.
+If the module is polled by the host with big frequency, that is when the :c:struct:`hid_report_sent_event` is received often by the module, the generated :c:struct:`motion_event` may contain value of ``0`` for both X and Y axes.
+
+.. note::
+   A HID report with reporting motion of ``0`` for both axes may be ignored by host computer's tools analyzing the HID report rate.
+
+   Increase value of the used scale factor or reduce time for transition between two points in the trajectory to increase the generated motion values.
```
Comment (kapi-no) [inline applications/nrf_desktop/doc/motion.rst:188]: add reference to KConfig option for the scale factor

Comment (MarekPieta) [reply applications/nrf_desktop/doc/motion.rst:188]: Added a generic note that links to module's configuration section (it seems more suitable after I applied reword)

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -206,13 +206,17 @@ nRF Machine Learning (Edge Impulse)
...
-  * :ref:`nrf_desktop_motion` - The :ref:`CONFIG_DESKTOP_MOTION_PM_EVENTS <config_desktop_app_options>` Kconfig option.
-  * :ref:`nrf_desktop_ble_latency` - The :ref:`CONFIG_DESKTOP_BLE_LATENCY_PM_EVENTS <config_desktop_app_options>` Kconfig option.
+    * :ref:`nrf_desktop_board` - The :ref:`CONFIG_DESKTOP_BOARD_PM_EVENTS <config_desktop_app_options>` Kconfig option.
+    * :ref:`nrf_desktop_motion` - The :ref:`CONFIG_DESKTOP_MOTION_PM_EVENTS <config_desktop_app_options>` Kconfig option.
+    * :ref:`nrf_desktop_ble_latency` - The :ref:`CONFIG_DESKTOP_BLE_LATENCY_PM_EVENTS <config_desktop_app_options>` Kconfig option.
 
-  All listed Kconfig options are enabled by default and depend on the :kconfig:option:`CONFIG_CAF_PM_EVENTS` Kconfig option.
+    All listed Kconfig options are enabled by default and depend on the :kconfig:option:`CONFIG_CAF_PM_EVENTS` Kconfig option.
+  * Kconfig option to configure motion generated per second during a button press in :ref:`nrf_desktop_motion` implementation that generates motion from button presses (:ref:`CONFIG_DESKTOP_MOTION_BUTTONS_MOTION_PER_SEC <onfig_desktop_app_options>`).
```
Comment (mkapala-nordic) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:218]:
```suggestion
  * Kconfig option to configure motion generated per second during a button press in :ref:`nrf_desktop_motion` implementation that generates motion from button presses (:ref:`CONFIG_DESKTOP_MOTION_BUTTONS_MOTION_PER_SEC <config_desktop_app_options>`).
```

File: applications/nrf_desktop/doc/motion.rst [hunk 5]
```diff
@@ -106,10 +132,18 @@ In these configurations, the module is a configuration channel listener and it p
 Implementation details
 **********************
 
+Implementation details depend on the selected implementation.
+See the following sections for details specific to a given implementation.
+
+The last section is common for all of the implementations and contains information useful for HID report rate measurements.
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/doc/motion.rst:138]: Sure

File: applications/nrf_desktop/doc/motion.rst [hunk 6]
```diff
@@ -141,12 +175,55 @@ Upon connection, the following happens:
...
+#. At this point, a next motion sampling is performed and the next :c:struct:`motion_event` sent.
 
 The module continues to sample data until disconnection or when there is no motion detected.
 The ``motion`` module assumes no motion when a number of consecutive samples equal to :ref:`CONFIG_DESKTOP_MOTION_SENSOR_EMPTY_SAMPLES_COUNT <config_desktop_app_options>` returns zero on both axis.
 In such case, the module will switch back to ``STATE_IDLE`` and wait for the motion sensor trigger.
+
+Movement data from buttons
+==========================
+
+Motion is generated from measured button press times.
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/doc/motion.rst]: Seems ok, I added one more sentence of introduction to make it more clear.

File: applications/nrf_desktop/doc/motion.rst [hunk 7]
```diff
@@ -141,12 +175,55 @@ Upon connection, the following happens:
...
+The time measurements rely on the hardware clock.
+If available, the module uses the :c:func:`k_cycle_get_64` function to read the hardware clock.
+Otherwise, the :c:func:`k_cycle_get_32` function is used for that purpose.
+
+When an HID subscriber is connected, that is when the device is connected either over USB or Bluetooth LE, the module forwards the generated motion to subscriber using :c:struct:`motion_event`.
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/doc/motion.rst]: Aligned to new version in a separate commit. Please check

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 4 above)
Comment (MarekPieta) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: Slightly rephrased

File: applications/nrf_desktop/doc/motion.rst [hunk 8]
```diff
@@ -205,3 +207,25 @@ Otherwise, the :c:func:`k_cycle_get_32` function is used for that purpose.
...
 The subsequent :c:struct:`motion_event` is submitted when the previously generated motion data is sent to the subscriber, that is when :c:struct:`hid_report_sent_event` is received by the module.
 The values of motion for X and Y axes are generated from an updated timestamp and the previously reported position.
+
+.. _nrf_desktop_motion_report_rate:
+
+Motion report rate
+==================
+
+If the device is polled for HID data with big frequency, that is when the :c:struct:`hid_report_sent_event` is received often by the module, the generated :c:struct:`motion_event` may contain value of ``0`` for both X and Y axes.
+A HID report with reporting motion of ``0`` for both axes may be ignored by host computer's tools that analyze the HID report rate.
```
Comment (kapi-no) [inline applications/nrf_desktop/doc/motion.rst:217]: This one looks like a note/caution message

File: applications/nrf_desktop/doc/motion.rst [hunk 9]
```diff
@@ -205,3 +207,25 @@ Otherwise, the :c:func:`k_cycle_get_32` function is used for that purpose.
 When a HID subscriber connects, that is when the device is connected either over USB or Bluetooth LE, the module submits the first :c:struct:`motion_event` containing motion calculated from the previously reported position.
 The subsequent :c:struct:`motion_event` is submitted when the previously generated motion data is sent to the subscriber, that is when :c:struct:`hid_report_sent_event` is received by the module.
 The values of motion for X and Y axes are generated from an updated timestamp and the previously reported position.
+
+.. _nrf_desktop_motion_report_rate:
+
+Motion report rate
+==================
+
+If the device is polled for HID data with big frequency, that is when the :c:struct:`hid_report_sent_event` is received often by the module, the generated :c:struct:`motion_event` may contain value of ``0`` for both X and Y axes.
```
Comment (kapi-no) [inline applications/nrf_desktop/doc/motion.rst:216]: high frequency*

**Review by kapi-no — APPROVED** [review]

**Review by annwoj — APPROVED** [review]
